### ISSUE TITLE
check_icmp uses inet_addr instead of inet_pton in function static void set_source_ip

### ISSUE THREAD

**Issue by jhaprins** [nagios-plugins/nagios-plugins#666]
The function set_source_ip in check_icmp is using the IPv4 only function inet_addr to determine if the provided address is a valid address or if the address should be converted from an interface name to an address. This works fine for IPv4 addresses, but in an IPv6 configuration this fails with the messages: Cannot determine IP address of interface 2001:db8::1: Inappropriate ioctl for device

This is probably an oversight from the time when IPv6 support was added to check_icmp. Everywhere else in the code inet_pton is used and the address_family specific things are added.

**Comment by sawolf** [comment]
From your description, I can probably get the patch sorted out myself, but we still don't have IPv6 support around here and I won't be able to test it. I'll post in here again closer to the plugins 2.4.1 release date if you're willing to test the patch.

**Comment by jhaprins** [comment]
Hello,
I just now noticed your patch and the availability in master.
I'm going to test it somewhere the coming days.
Jan Hugo Prins

**Comment by jhaprins** [comment]
I have quickly tested your patch agains 2.3.3 that is in EPEL for Centos 7. This gives some errors again:[root@bgp02 plugins]# ./check_icmp -6 -H ping.ripe.net -s 2001:db8:0:520::6
check_icmp: Cannot bind to IP address 2001:db8:0:520::6: Cannot assign requested address
[root@bgp02 plugins]# ./check_icmp -6 -H ping.ripe.net -s 10ge6
check_icmp: Cannot determine IP address of interface 10ge6: Inappropriate ioctl for device

I'm going to test version 2.4.2 later to see if that works out of the box.

**Comment by jhaprins** [comment]
Hello,
I have partly good news. 
I have been able to build 2.4.2 for Centos7 and was able to test your patch. 
The result is twofold. 
If I set a source IPv6 address everything works fine now.
When I set a source interface I still get an error. The -s option tells me that I should be able to set an interface.[root@bgp02 plugins]# ./check_icmp -6 -H ping.ripe.net -s 2001:db8:6::2:e
OK - ping.ripe.net rta 3.451ms lost 0%|rta=3.451ms;200.000;500.000;0; pl=0%;40;80;0;100 rtmax=3.593ms;;;; rtmin=3.192ms;;;; 
[root@bgp02 plugins]# ./check_icmp -6 -H ping.ripe.net -s 2001:db8:0:520::3
OK - ping.ripe.net rta 3.980ms lost 0%|rta=3.980ms;200.000;500.000;0; pl=0%;40;80;0;100 rtmax=4.168ms;;;; rtmin=3.672ms;;;; 
[root@bgp02 plugins]# ./check_icmp -6 -H ping.ripe.net -s 10ge6
check_icmp: Cannot determine IP address of interface 10ge6: Inappropriate ioctl for device
